executor/adapter.go

```diff
@@ -259,9 +260,6 @@ func (a *ExecStmt) handleNoDelayExecutor(ctx context.Context, sctx sessionctx.Co
 
 	var err error
 	defer func() {
-		if pi != nil {
```

Comment:
Seems `pi` is not used in this function?

Reply:
`pi` in `handleNoDelayExecutor` and `NewChunk()` is working for set command to `ComSleep` after sql execute. I delete both of them and move to here: https://github.com/pingcap/tidb/pull/7844/files/912d55e6d9b185ee5929159641dd8ee32a475053#diff-dc417f5322100e2e83a8818c70a25878R593